src/main/java/net/minecraftforge/common/util/MutableHashedLinkedMap.java

```diff
@@ -249,15 +249,13 @@ private void remove(Entry e)
         else if (e.previous != null) // Should never be null, but just in case.
         {
             e.previous.next = e.next;
-            e.previous = null;
         }
 
         if (last == e)
             last = e.previous;
         else if (e.next != null) // Should never be null, but just in case.
         {
             e.next.previous = e.previous;
-            e.next = null;
```

Comment:
So, in general, this is not how you fix this. First, you are correct that the implementation is not correct as previous is set to null before being properly linked in the next section. However, removing those checks would leave hanging references which the garbage collector would never be able to handle.

The way to fix this would be to set these values to null at the end of the remove section. Though I have no idea whether this actually fixes the issue since I haven't tested yet.

Reply:
Ok I'll try to add them to the bottom of the section and see what happens

Reply:
So i've tried moving the null assignments out of the remove section (either at the bottom or out of the function call) but that reintroduces the bug. I also don't understand why we set `entry.previous.next` if we are setting `entry.previous` to null anyway